### PR TITLE
Feat/home

### PR DESCRIPTION
**Pull request by bhavyaj12**
### Added Home Page complete

- Added carousel, nav top, footer, category list (components)
- Added state and context for categories, and axios call to fetch categories
- Added category data
- Added context, localStorage for theme

**Please check app is RESPONSIVE and with theme change functionality**

### What not to check
- Nav Links
- Any other pages and routes
- Any functionality

Screenshot (109)
Screenshot (111)

Screenshot (113)
Screenshot (114)
Screenshot (116)

### CODE REVIEW

File: src/components/CatergoryList/CategoryList.jsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { useCategory } from "contexts";
+import "./category-list.css";
+
+const CategoryList = () => {
+  const { categoryState } = useCategory();
+  console.log(categoryState);
+  return (
+    <div className="categories-container mt-7">
+      {categoryState.categories.map((category) => {
```
Comment (sashaboi) [inline src/components/CatergoryList/CategoryList.jsx:9]: You could add a loader here till the category state has items to get better user experience.

File: src/components/HomeCarousel/HomeCarousel.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  {
+    id: "aco",
+    img: "assets/banner-3.jpg",
+  },
+];
+
+const HomeCarousel = () => {
+  return (
+    <Carousel autoPlay infiniteLoop showThumbs={false}>
+      {carouselImg.map((banner) => {
```
Comment (sashaboi) [inline src/components/HomeCarousel/HomeCarousel.jsx:24]: A loader here as well maybe ?

File: src/components/NavigationTop/NavigationTop.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  useEffect(() => {
+    localStorage.setItem("halcyon-gaming-theme", theme);
+  }, [theme]);
+
+  return (
+    <header id="header">
+      <nav className="store-nav-bar">
+        <div className="store-logo-box">
+          {theme === "dark" ? (
```
Comment (sashaboi) [inline src/components/NavigationTop/NavigationTop.jsx:22]: loved that you were able to get dark theme up and running , good job!

File: src/components/index.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import NavigationTop from "./NavigationTop/NavigationTop";
```
Comment (sashaboi) [inline src/components/index.js:1]: these make code cleaner , good job on using these.

File: src/routes/Routes.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Routes, Route } from 'react-router-dom';
+import { HomePage } from 'pages';
+import Mockman from 'mockman-js';
+
+const SiteRoutes = () => {
+    return (
+        <Routes>
+            <Route path="/" element={<HomePage />} />
```
Comment (sashaboi) [inline src/routes/Routes.jsx:8]: Do put a note for later down the road to add the 404 route as well.

File: src/utilities/fetchCategories.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import axios from "axios";
+
+const fetchCategories = () => axios.get('/api/categories');
```
Comment (sashaboi) [inline src/utilities/fetchCategories.js:3]: this needs error handling ! 
Rn the server is in your control but later down the road you have to be ready if the server doesnt work , please use try catch / axios error handling to make code robust.

**Review by sashaboi — APPROVED** [review]
The layout looks amazing Bhavya , you have a consistent theme going love that. 

I have a made a few ( optional ) suggestions below, please make a note of them as you go on developing the app ahead. 

One of the higher priority changes might be to handle errors in your api call, hope that gets dealth with earlier than later. 

good job on the styling , theme, DARK mode (wow) , looking forward for next versions of the project !

File: src/utilities/fetchCategories.js (hunk 6 above)
Comment (bhavyaj12) [reply src/utilities/fetchCategories.js:3]: So I'm handling the error where I'm calling this function, using async await there so I could show that error using react toastify, will add it

**Pull request merged**